flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java

```diff
@@ -101,11 +108,28 @@ static CatalogLoader createCatalogLoader(String name, Map<String, String> proper
     }
   }
 
+  @Override
+  public String factoryIdentifier() {
+    return IDENTIFIER;
+  }
+
+  @Override
+  public Set<ConfigOption<?>> requiredOptions() {
+    return Sets.newHashSet();
+  }
+
+  @Override
+  public Set<ConfigOption<?>> optionalOptions() {
+    final Set<ConfigOption<?>> options = Sets.newHashSet();
+    options.add(FactoryUtil.PROPERTY_VERSION);
+    return options;
+  }
+
   @Override
   public Map<String, String> requiredContext() {
     Map<String, String> context = Maps.newHashMap();
-    context.put(CatalogDescriptorValidator.CATALOG_TYPE, "iceberg");
-    context.put(CatalogDescriptorValidator.CATALOG_PROPERTY_VERSION, "1");
+    context.put(TYPE, "iceberg");
+    context.put(PROPERTY_VERSION, "1");
```

Comment:
I don't think that this config value should be required anymore (as people should be allowed to create V2 tables).

Reply:
@kbendick this `property-version` is indicating the flink connector's properties version, not the apache iceberg table format,  which is used for guarantee the properties compatibility.